test/gtest/layout_transpose.cpp

```diff
@@ -232,20 +232,9 @@ struct transpose_dims
     }
 };
 
-struct transpose_invoke_param : public miopen::InvokeParams
-{
-    ConstData_t src = nullptr;
-    Data_t dst      = nullptr;
-
-    transpose_invoke_param(ConstData_t src_, Data_t dst_) : src(src_), dst(dst_) {}
-    transpose_invoke_param(miopen::InvokeType type_, ConstData_t src_, Data_t dst_)
-        : InvokeParams{type_}, src(src_), dst(dst_)
-    {
-    }
-
-    Data_t GetWorkspace() const { return nullptr; }
-    std::size_t GetWorkspaceSize() const { return 0; }
-};
+template <typename T>
+static auto gen_value =
+    [](auto... is) { return static_cast<T>(prng::gen_A_to_B(RAND_INTEGER_MIN, RAND_INTEGER_MAX)); };
```

Comment:
convert to a function please.

Reply:
something like: 
```
template <typename T, typename... Args>
T genValue(Args... args) {
   return something;
}
```

Reply:
We can keep `genValue` as an object of class with template `operator()` if we need to prevent `genValue` from participating in ADL. @randyspauldingamd was it the intention?

Reply:
Actually, I realize that `gen_value` needs to be a functor, so let's leave it as a lambda. @scerzh , your suggestion also means the same thing but more typing, so lambda offers the lazy way out.

Reply:
This is only used in tensor::generate, and I don't see any use cases taking a function. Exploratory work is out of scope for this PR, how about we discuss this as future work?